content/shmem_scan.tex

```diff
@@ -0,0 +1,121 @@
+\end{Csynopsis}
+where \TYPE{} is one of the integer, real, or complex types supported
+for the SUM operation and has a corresponding \TYPENAME{} as specified
+by Table \ref{teamreducetypes}.
+
+\begin{apiarguments}
+  \apiargument{IN}{team}{
+    The team over which to perform the operation.
+  }
+  \apiargument{OUT}{dest}{
+    Symmetric address of an array, of length \VAR{nreduce} elements,
+    to receive the result of the scan routines.  The type of
+    \dest{} should match that implied in the SYNOPSIS section.
+  }
+  \apiargument{IN}{source}{
+    Symmetric address of an array, of length \VAR{nreduce} elements,
+    that contains one element for each separate scan routine.
+    The type of \source{} should match that implied in the SYNOPSIS
+    section.
+  }
+  \apiargument{IN}{nreduce}{
+    The number of elements in the \dest{} and \source{} arrays.
+  }
+\end{apiarguments}
+
+\apidescription{
+
+  The \FUNC{shmem\_sum\_inscan} and \FUNC{shmem\_sum\_exscan} routines
+  are collective routines over an \openshmem team that compute one or
+  more scan (or prefix sum) operations across symmetric arrays on
+  multiple \acp{PE}.  The scan operations are performed with the SUM
+  operator.
+
+  The \VAR{nreduce} argument determines the number of separate scan
+  operations to perform.  The \source{} array on all \acp{PE}
+  participating in the operation provides one element for each scan.
+  The results of the scan operations are placed in the \dest{} array
+  on all \acp{PE} participating in the scan.
+
+  The \FUNC{shmem\_sum\_inscan} routine performs an inclusive scan
+  operation, while the \FUNC{shmem\_sum\_exscan} routine performs an
+  exclusive scan operation.  
+
+  For \FUNC{shmem\_sum\_inscan}, the value of the $j$-th element in
+  the \VAR{dest} array on \ac{PE}~$i$ is defined as:
+  \begin{equation*}
+    \textrm{dest}_{i,j} = \displaystyle\sum_{k=0}^{i} \textrm{source}_{k,j}
+  \end{equation*}
+
+  For \FUNC{shmem\_sum\_exscan}, the value of the $j$-th element in
+  the \VAR{dest} array on \ac{PE}~$i$ is defined as:
+  \begin{equation*}
+    \textrm{dest}_{i,j} =
+    \begin{cases}
+    \displaystyle\sum_{k=0}^{i-1} \textrm{source}_{k,j}, & \text{if} \; i \neq 0 \\
+    0,  & \text{if} \; i = 0
+    \end{cases}
+  \end{equation*}
+
+  The \source{} and \dest{} arguments must either be the same
+  symmetric address, or two different symmetric addresses
+  corresponding to buffers that do not overlap in memory. That is,
+  they must be completely overlapping or completely disjoint.
+
+  Team-based scan routines operate over all \acp{PE} in the provided
+  team argument. All \acp{PE} in the provided team must participate in
+  the scan operation.  If \VAR{team} compares equal to
+  \LibConstRef{SHMEM\_TEAM\_INVALID} or is otherwise invalid, the
+  behavior is undefined.
+
+  Before any \ac{PE} calls a scan routine, the \dest{} array on all
```

Comment:
In response to the message in the example section below:
By omission, this is saying that the src arrays on all pes does not need to be ready.